src/equations/compressible_euler_2d.jl

```diff
@@ -809,6 +809,93 @@ end
     return SVector(f1m, f2m, f3m, f4m)
 end
 
+"""
+    FluxLMARS(c)(u_ll, u_rr, orientation_or_normal_direction,
+                 equations::CompressibleEulerEquations2D)
+
+Low Mach number approximate Riemann solver (LMARS) for atmospheric flows using
+an estimate `c` of the speed of sound.
+
+References:
+- Xi Chen et al. (2013)
+  A Control-Volume Model of the Compressible Euler Equations with a Vertical
+  Lagrangian Coordinate
+  [DOI: 10.1175/MWR-D-12-00129.1](https://doi.org/10.1175/mwr-d-12-00129.1)
+"""
+struct FluxLMARS{SpeedOfSound}
+    # Estimate for the speed of sound
+    speed_of_sound::SpeedOfSound
+end
+
+@inline function (flux_lmars::FluxLMARS)(u_ll, u_rr, orientation::Integer,
+                                         equations::CompressibleEulerEquations2D)
+    c = flux_lmars.speed_of_sound
+
+    # Unpack left and right state
+    rho_ll, v1_ll, v2_ll, p_ll = cons2prim(u_ll, equations)
+    rho_rr, v1_rr, v2_rr, p_rr = cons2prim(u_rr, equations)
+
+    if orientation == 1
+        v_ll = v1_ll
+        v_rr = v1_rr
+    else # orientation == 2
+        v_ll = v2_ll
+        v_rr = v2_rr
+    end
+
+    rho = 0.5 * (rho_ll + rho_rr)
+    p = 0.5 * (p_ll + p_rr) - 0.5 * c * rho * (v_rr - v_ll)
+    v = 0.5 * (v_ll + v_rr) - 1 / (2 * c * rho) * (p_rr - p_ll)
+
+    if v >= 0
+        f1, f2, f3, f4 = v * u_ll
+    else
+        f1, f2, f3, f4 = v * u_rr
+    end
+
+    if orientation == 1
+        f2 += p
+    else # orientation == 2
+        f3 += p
+    end
+    f4 += p * v
+
+    return SVector(f1, f2, f3, f4)
+end
+
+@inline function (flux_lmars::FluxLMARS)(u_ll, u_rr, normal_direction::AbstractVector,
+                                         equations::CompressibleEulerEquations2D)
+    c = flux_lmars.speed_of_sound
+
+    # Unpack left and right state
+    rho_ll, v1_ll, v2_ll, p_ll = cons2prim(u_ll, equations)
+    rho_rr, v1_rr, v2_rr, p_rr = cons2prim(u_rr, equations)
+
+    v_ll = v1_ll * normal_direction[1] + v2_ll * normal_direction[2]
+    v_rr = v1_rr * normal_direction[1] + v2_rr * normal_direction[2]
+
+    # Note that this is the same as computing v_ll and v_rr with a normalized normal vector
+    # and then multiplying v by `norm_` again, but this version is slightly faster.
+    norm_ = norm(normal_direction)
+
+    rho = 0.5 * (rho_ll + rho_rr)
+    p = 0.5 * (p_ll + p_rr) - 0.5 * c * rho * (v_rr - v_ll) / norm_
+    v = 0.5 * (v_ll + v_rr) - 1 / (2 * c * rho) * (p_rr - p_ll) * norm_
+
+    if v >= 0
+        f1, f2, f3, f4 = u_ll * v
+        # f4 += p_ll * v ???
+    else
+        f1, f2, f3, f4 = u_rr * v
+        # f4 += p_rr * v ???
+    end
```

Comment:
?

Reply:
This is the point I feel unconfident with.

The paper by Chen et al. derives the fluxes for potential temperature as the fourth conserved quantity, not for total energy. By analogy, I think the current implementation is correct. It would also be consistent to what we currently have in `compressible_euler_3d.jl`. 

But: The implementation in the master's thesis and in Oswald Knoth's PR is different (these are the commented lines)
https://github.com/NichtLucas/Trixi.jl/blob/60d43aaa5d9f7b0b230a555a0e25ad6acc7619e7/src/equations/compressible_moist_euler_2d.jl#L473
https://github.com/OsKnoth/Trixi.jl/blob/369a8876d2585ae3c66864c2934fe0d67957660c/src/equations/2d/compressible_euler.jl#L800

Reply:
Thank you @andrewwinters5000 @DanielDoehring @ranocha for your help!